### PR TITLE
[WIP] feat: import libsvm

### PR DESCRIPTION
**Pull request by WenheLI**

### CODE REVIEW

**Comment by WenheLI** [comment]
Overall, the implementation is there. However, I think there is some numerical issue about libsvm-wasm. Right now, we could review the implementation on scikit side.

**Review by dcrescim — APPROVED** [review]
@WenheLI This is amazing, dude! I love the port of LibSVM to wasm. I think this is gonna be a game-changer.

I threw most of my notes below. There's some simple stuff (we need to run prettier.js), and some possible reformatting just to make the code tighter in certain places, but it's ready :smile:

I had two overarching questions that we can talk through tomorrow.
1. Is there a way to make the `predict` function synchronous? All of the `predict` functions for every other Estimator in this library are synchronous, and I think people will be confused if they have to `await` only the `predict` for SVM/SVR's?
2. For the `fit` function, we convert our input into a Tensor (which basically does a data copy, if they pass in an array), and then we convert that back into an array. Is there a way that we can simply keep it in array form the whole time (I think that will speed this up)? If we can only pass arrays into the `feedSamples` function of LibSVM then we might as well keep it as an array the whole time.

This is amazing work dude.

File: shared/lib/svm/SVC.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        degree = 3,
+        gamma = 'scale',
+        coef0 = 0,
+        tol = 1e-3,
+        C = 1,
+        epsilon = 0.1,
+        shrinking = true,
+        cacheSize = 200,
+        classWeight = {},
+        maxIter = -1
```
[the remark this answers is not in the document]
Comment (WenheLI) [reply shared/lib/svm/SVC.ts:38]: For now, we do not need them. But we could support them later

**Comment by WenheLI** [comment]
One quick question, how to run the linter. I did not see any command here

**Comment by dcrescim** [comment]
This PR looks great @WenheLI ! I think your editor isn't using the prettier.js config in the `.prettierrc` in the root of the directory. There are `npm run lint` commands in both the scikitjs-node, scikitjs-browser, but the easiest way to test the whole repo is just to run `npm run test:browser` and `npm run test:node` in the root directory. Gimme a ring and I'll show you how to set prettier.js to run on file save in your editor.

**Comment by dcrescim** [comment]
Also you can run prettier.js from the command line to fix any issues that it finds. Just do

`npx prettier --write shared/*`

**Pull request merged**